spec/electric_slide/call_queue_spec.rb

```diff
@@ -39,4 +39,39 @@
   it "should raise when given an invalid Agent" do
     expect { queue.add_agent nil }.to raise_error
   end
+
+  describe '#call_agent' do
+    let(:queue) { ElectricSlide::CallQueue.new(connection_type: :call) }
+    let(:agent) { ElectricSlide::Agent.new }
+    let!(:agent_call) { Adhearsion::OutboundCall.new }
+
+    before do
+      allow(Adhearsion::OutboundCall).to receive(:new) { agent_call }
+      allow(agent).to receive(:dial_options_for) {
+        { confirm: double('ConfirmController') }
+      }
+
+      allow(call_a).to receive(:active?) { true }
+      allow(agent_call).to receive(:dial)
+      queue.connect(agent, call_a)
+    end
+
+    it "sets the agent's `call` attribute" do
+      expect(agent.call).to be agent_call
+    end
+
+    context 'when the call ends' do
+      def hangup(call)
+        # @fixme Is there a better way to hang up to test `on_end` handlers?
+        require 'punchblock'
```

Comment:
Any require statement like this should be done at the top of the file rather than inside a method.